### PR TITLE
Add kernel release check to environment_check.sh

### PR DESCRIPTION
**Pull request by PhanLe1010**
longhorn/longhorn#6854

### CODE REVIEW

**Comment by PhanLe1010** [comment]
@mergify backport v1.5.x v1.4.x

**Comment by PhanLe1010** [comment]
ping @derekbit @innobead

File: scripts/environment_check.sh [hunk 1]
```diff
@@ -273,6 +273,28 @@ check_nodes() {
...
+verlt() {
+    ! verlte "$2" "$1"
+}
+
+check_kernel_release() {
+  local pod=$1
+
+  recommended_kernel_release="5.8"
+
+  local kernel=$(detect_node_kernel_release ${pod})
```
Comment (james-munson) [inline scripts/environment_check.sh:289]: shellcheck would be happier with a quoted "${pod}" both here and below, but not a big deal.

**Review by james-munson — APPROVED** [review]
LGTM.  This will be good to have in place.

File: scripts/environment_check.sh (hunk 1 above)
Comment (PhanLe1010) [reply scripts/environment_check.sh:289]: Looks like we are not using quote in a lot of places in this script already so perhaps we should keep the same to keep it uniform unless we decided to change all of them in another ticket?

**Review by ejweber — APPROVED** [review]
In general, LGTM, but one minor concern that we discussed offline.

It may be difficult to remember to keep this 5.8 updated. I'm somewhat fine with relying on developers to search the code base for any uses of some constant when updating it, but this one might be problematic, as we document our recommendation in a different repo. Hopefully we don't change our recommendation often and we can keep on eye on it.

**Pull request merged**